parsec/core/mountpoint/manager.py

```diff
@@ -35,7 +35,7 @@
 
 RunnerType = Callable[
     [UserFS, WorkspaceFS, Path, dict[Any, Any], EventBus],
-    _AsyncGeneratorContextManager[Path],
+    _AsyncGeneratorContextManager[Union[Path, PurePath]],
```

Comment:
```suggestion
    _AsyncGeneratorContextManager[Path | PurePath],
```

Use `|` that way, we don't have to include `Union`

Reply:
`Union` is unfortunately mandatory here. Mypy seems to require it.
Here are the errors I get with the `|` syntax.
```
parsec/core/mountpoint/manager.py:38: error: Type expected within [...]  [misc]
parsec/core/mountpoint/manager.py:38: error: The type "Type[_AsyncGeneratorContextManager[Any]]" is not generic and not indexable  [misc]
parsec/core/mountpoint/manager.py:38: error: Unsupported left operand type for | ("Type[Path]")  [operator]
```